app/code/Magento/CatalogInventory/Model/Indexer/ProductPriceIndexFilter.php

```diff
@@ -104,6 +104,14 @@ public function modifyPrice(IndexTableStructure $priceTable, array $entityIds =
             $select->where('stock_item.use_config_manage_stock = 0 AND stock_item.manage_stock = 1');
         }
 
+        if ($entityIds !== null) {
+            if (count($entityIds) > 1) {
```

Comment:
Why we should count elements here? $entityIds always array, let's use `$select->where('stock_item.product_id in (?)', $entityIds)` in all cases if array is not empty

Reply:
Yeah fair point I copied the implementation from here `Magento\Catalog\Model\ResourceModel\Product\Indexer\Price\Query\BaseFinalPrice::getQuery`